waspc/data/Generator/templates/server/src/auth/providers/email/resetPassword.ts

```diff
@@ -37,10 +36,7 @@ export async function resetPassword(
             hashedPassword: password,
         });
     } catch (e) {
-        const reason = e.name === tokenVerificationErrors.TokenExpiredError
-            ? 'expired'
-            : 'invalid';
-        throw new HttpError(400, `Password reset failed, ${reason} token`);
+        throw new HttpError(400, `Password reset failed, invalid token`);
```

Comment:
Simplifying the error messages (in part to avoid parsing `oslo/jwt` messages since they are just `Error`s)

Reply:
Aha, but now we can't say if it is expired or something else, right? At least we are not doing it any more. Is that bad?

Can you at least check here if `e` was indeed thrown by `oslo/jwt`?
It is generally a bit scary to catch just about any error and then ignore its value, as it could be anything really -> maybe `createProviderId` failed, or `findAuthIdentity` failed -> yeah, this is too wide a net we are casting here.

I would both make the `try` capture less of the code (just `validateJWT` statement) and also try to look a bit at the error, and throw 400 only if it is what we expected, otherwise rethrow it.

Reply:
> It is generally a bit scary to catch just about any error and then ignore its value, as it could be anything really -> maybe createProviderId failed, or findAuthIdentity failed -> yeah, this is too wide a net we are casting here.

I wouldn't go any more specific than "something failed" or "the token is invalid" because that gives information to the attacker e.g. if we said "that user cannot be found" that would reveal extra info.

I proposed that we:
- say "the token is invalid" in the case of an invalid token (code: 400)
- say "something went wrong" in any other case (code: 500)

Reply:
Updated error handling here: https://github.com/wasp-lang/wasp/pull/1852/commits/fa0ad66d4b789df5f3c053933a1271bc7aaac58e

Reply:
Yeah exactly -> this is default behaviour. So if you throw a HttpError, it will go out with that status, but if you just rethrow, our default error middleware we wrap it into 500, and hide any info. And it will show more info in development (I think, if not we should add that, but that is nice property of having it centralized like that).

Reply:
Maybe then best to remove the try catch if we handled some of the specific cases by throwing `HttpError` 😄 

Okay, I'll try to write it like that and report back.